mvrx/proguard-rules.pro

```diff
@@ -34,6 +34,7 @@
 
 # Members of the Kotlin data classes used as the state in MvRx are read via Kotlin reflection which cause trouble
 # with Proguard if they are not kept.
--keepclassmembers class ** implements com.airbnb.mvrx.MvRxState {
+# During reflection cache warming also the types are accessed via reflection. Need to keep them too.
+-keepclassmembers,includedescriptorclasses class ** implements com.airbnb.mvrx.MvRxState {
```

Comment:
First time I see descriptorclasses modifier. Is that the magic comes from? 😊

Reply:
Yes, read: 

> **includedescriptorclasses**
> Specifies that any classes in the type descriptors of the methods and fields that the -keep option keeps should be kept as well. This is typically useful when keeping native method names, to make sure that the parameter types of native methods aren't renamed either. Their signatures then remain completely unchanged and compatible with the native libraries.

from: https://www.guardsquare.com/en/products/proguard/manual/usage

and keeping the type is exactly what is needed here.